### DISCUSSION TITLE
Lost tag in Scenario Outline

### DISCUSSION THREAD

**Discussion by nicolocastro89** [behave/behave#947] [Q&A] [answered]
Hi everyone,
I'll preface this by saying that I realize this is the expected behaviour but I would like to know if someone can help me with a workaround.
I am testing different versions of a product, and some tests are only supposed to run up to a certain version, while others only starting from a certain version. In order to implement this I added tags to the scenarios to indicate for which versions they should run. I then run the tests using -D version=vX, and placed some login in the _before_scenario_ hook to compare the provided version with the one specified in the tag (if present), in particular I am using standard version notations, so for example a scenario might have as a tag '@version=">=v3"'. This worked perfectly in the Scenarios, but as you might have spotted this does not work well in the Scenario Outlines, as the make_row_tags method of the ScenarioOutlineBuilder class interprets < and > as placeholders (most of the times only one is present btw) and skips the tag. I could replace the symbols with the string equivalent (gt, lt, ge, le, eq) but it would make everything less readable. Does anyone have any other ideas?

**Comment by jenisys** [comment, marked as answer]
You are currently relying on undefined behavior.

A ScenarioOutline is basically a template that contains `<placeholder>`.
Because tags of a ScenarioOutline can also contain placeholders, some characters are currently rejected when the template is instantiated. Therefore, you must currently use a solution that overcomes this limitation.

**Reply by nicolocastro89** [reply]
Thanks. Yeah i realized it was a bit of a weird request, however i also noticed that quotation marks are also removed from tags in scenario outlines, and this is a bit surprising at it is a big difference from scenarios, and i can't justify it like the  angular brackets. Is there any reason for that?
